source/tests/SGM/SGMEnergyTrajectoryTestView.cpp

```diff
@@ -242,34 +263,20 @@ void SGMEnergyTrajectoryTestView::setupUi()
 	energyVsTimeSeries->setDescription("Energy produced by constant velocity grating angle");
 }
 
-void SGMEnergyTrajectoryTestView::setTheoreticalPlotData(SGMGratingSupport::GratingTranslation gratingTranslation)
+void SGMEnergyTrajectoryTestView::setTheoreticalPlotData(SGMGratingSupport::GratingTranslation gratingTranslation,
+                                                         SGMUndulatorSupport::UndulatorHarmonic undulatorHarmonic)
 {
 	if(gratingTranslation != SGMGratingSupport::UnknownGrating) {
-		double startValue;
-		double endValue;
-
-		switch(gratingTranslation) {
-		case SGMGratingSupport::LowGrating:
-			startValue = 200;
-			endValue = 800;
-			break;
-		case SGMGratingSupport::MediumGrating:
-			startValue = 400;
-			endValue = 1400;
-			break;
-		case SGMGratingSupport::HighGrating:
-			startValue = 600;
-			endValue = 2500;
-			break;
-		default:
-			startValue = 0;
-			endValue = 0;
-		}
+		double startValue = 200;
+		double endValue = 3000;
 
 		double increment = 1;
 
 		int numberOfItems = int((endValue - startValue) / increment);
```

Comment:
I know that this line isn't part of the pull request, but I think I must be missing something here.  There is the following code:

```c++
double increment = 1;
int numberOfItems = int(endValue-startValue)/increment);
```

Am I missing something and increment is supposed to be updated somewhere?  Also, generally I think you usually want to add a +1 to that calculation when computing the count of a number of items.